HLTriggerOffline/Scouting/plugins/ScoutingEGammaCollectionMonitoring.cc

```diff
@@ -0,0 +1,459 @@
+                   "sctElectron_EBEB_appliedID_invMass_passDoubleEG_DST",
+                   400,
+                   0.,
+                   200.);
+  histos.sctElectron.h1InvMassIDEBEE_passDoubleEG_DST =
+      ibook.book1D("sctElectron_EBEE_appliedID_invMass_passDoubleEG_DST",
+                   "sctElectron_EBEE_appliedID_invMass_passDoubleEG_DST",
+                   400,
+                   0.,
+                   200.);
+  histos.sctElectron.h1InvMassIDEEEE_passDoubleEG_DST =
+      ibook.book1D("sctElectron_EEEE_appliedID_invMass_passDoubleEG_DST",
+                   "sctElectron_EEEE_appliedID_invMass_passDoubleEG_DST",
+                   400,
+                   0.,
+                   200.);
+
+  histos.sctElectron.h1InvMassID_passSinglePhoton_DST =
+      ibook.book1D("sctElectron_appliedID_invMass_passSinglePhoton_DST",
+                   "sctElectron_appliedID_invMass_passSinglePhoton_DST",
+                   400,
+                   0.,
+                   200.);
+  histos.sctElectron.h1InvMassIDEBEB_passSinglePhoton_DST =
+      ibook.book1D("sctElectron_EBEB_appliedID_invMass_passSinglePhoton_DST",
+                   "sctElectron_EBEB_appliedID_invMass_passSinglePhoton_DST",
+                   400,
+                   0.,
+                   200.);
+  histos.sctElectron.h1InvMassIDEBEE_passSinglePhoton_DST =
+      ibook.book1D("sctElectron_EBEE_appliedID_invMass_passSinglePhoton_DST",
+                   "sctElectron_EBEE_appliedID_invMass_passSinglePhoton_DST",
+                   400,
+                   0.,
+                   200.);
+  histos.sctElectron.h1InvMassIDEEEE_passSinglePhoton_DST =
+      ibook.book1D("sctElectron_EEEE_appliedID_invMass_passSinglePhoton_DST",
+                   "sctElectron_EEEE_appliedID_invMass_passSinglePhoton_DST",
+                   400,
+                   0.,
+                   200.);
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the
+// module  ------------
+void ScoutingEGammaCollectionMonitoring::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // The following says we do not know what parameters are allowed so do no
+  // validation Please change this to state exactly what you do use, even if it
+  // is no parameters
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("OutputInternalPath", "MY_FOLDER");
+  desc.add<edm::InputTag>("TriggerResultTag", edm::InputTag("TriggerResults", "", "HLT"));
+  desc.add<edm::InputTag>("ElectronCollection", edm::InputTag("slimmedElectrons"));
+  desc.add<edm::InputTag>("ScoutingElectronCollection", edm::InputTag("hltScoutingEgammaPacker"));
+  desc.add<edm::InputTag>("eleIdMapTight",
+                          edm::InputTag("egmGsfElectronIDs:cutBasedElectronID-RunIIIWinter22-V1-tight"));
+  descriptions.add("ScoutingEGammaCollectionMonitoring", desc);
+}
+
+bool ScoutingEGammaCollectionMonitoring::scoutingElectronID(const Run3ScoutingElectron el) const {
```

Comment:
if it's used in multiple class could be moved to a single utility file.

Reply:
@tihsu99: for future code developments, consider checking for usage of a single utility file. (See one more comment above.)